cli/azd/cmd/auth_token.go

```diff
@@ -40,26 +42,41 @@ func (f *authTokenFlags) Bind(local *pflag.FlagSet, global *internal.GlobalComma
 	f.global = global
 	output.AddOutputFlag(local, &f.outputFormat, []output.Format{output.JsonFormat}, output.NoneFormat)
 	local.StringArrayVar(&f.scopes, "scope", nil, "The scope to use when requesting an access token")
+	local.StringVar(&f.tenantID, "tenant-id", "", "The tenant id to use when requesting an access token.")
 }
 
 type authTokenAction struct {
-	credential azcore.TokenCredential
-	formatter  output.Formatter
-	writer     io.Writer
-	flags      authTokenFlags
+	credentialProvider func(context.Context, *auth.CredentialForCurrentUserOptions) (azcore.TokenCredential, error)
+	formatter          output.Formatter
+	writer             io.Writer
+	flags              authTokenFlags
 }
 
 func newAuthTokenAction(
-	credential azcore.TokenCredential,
+	authManager *auth.Manager,
 	formatter output.Formatter,
 	writer io.Writer,
 	flags authTokenFlags,
 ) *authTokenAction {
 	return &authTokenAction{
-		credential: credential,
-		formatter:  formatter,
-		writer:     writer,
-		flags:      flags,
+		credentialProvider: authManager.CredentialForCurrentUser,
+		formatter:          formatter,
+		writer:             writer,
+		flags:              flags,
+	}
+}
```

Comment:
Is there a use case where this is now called without a credential provider?  If not we should deprecate this version.

Reply:
The credential provider version was added to support some testing cases where we want to override the credential so we can mock and test some of the AuthToken action logic.  I took this approach vs making auth.Manager an interface since it felt a little cleaner.

`newAuthTokenAction` is still called by the generated code from wire to construct the action use the auth manager it constructs. 

I'll make the necessary changes such that wire pulls the CredentialProvider from the AuthManager and just calls this common constructor and then remove this one (I'll then shorten the name of `newAuthTokenActionWithCredentialProvider` back to `newAuthTokenAction`)